azurerm/resource_arm_network_security_group.go

```diff
@@ -37,6 +40,14 @@ func resourceArmNetworkSecurityGroup() *schema.Resource {
 				Type:     schema.TypeSet,
 				Optional: true,
 				Computed: true,
+				Set: func(v interface{}) int {
```

Comment:
Because the hash function only looks at `direction` and `priority` any other changes will be ignored by terraform.